starter-code/scripts/views/articleView.js

```diff
@@ -37,55 +38,57 @@
   articleView.handleFilters = function() {
     $('#filters').one('change', 'select', function() {
       var resource = this.id.replace('-filter', '');
       $(this).parent().siblings().children().val('');
       page('/' + resource + '/' +
       // Replace any/all whitespace with a '+' sign
-        $(this).val().replace(/\W+/g, '+')
-      );
+      $(this).val().replace(/\W+/g, '+')
+    );
     });
   };
 /* articleView.handleAuthorFilter = function() {
-     $('#author-filter').on('change', function() {
-       if ($(this).val()) {
-         $('article').hide();
-         $('article[data-author="' + $(this).val() + '"]').fadeIn();
-       } else {
-         $('article').fadeIn();
-         $('article.template').hide();
-       }
-       $('#category-filter').val('');
-     });
-   };
-
-   articleView.handleCategoryFilter = function() {
-     $('#category-filter').on('change', function() {
-       if ($(this).val()) {
-         $('article').hide();
-         $('article[data-category="' + $(this).val() + '"]').fadeIn();
-       } else {
-         $('article').fadeIn();
+$('#author-filter').on('change', function() {
+if ($(this).val()) {
+$('article').hide();
+$('article[data-author="' + $(this).val() + '"]').fadeIn();
+} else {
+$('article').fadeIn();
+$('article.template').hide();
+}
+$('#category-filter').val('');
+});
+};
+
+articleView.handleCategoryFilter = function() {
+$('#category-filter').on('change', function() {
+if ($(this).val()) {
+$('article').hide();
+$('article[data-category="' + $(this).val() + '"]').fadeIn();
+} else {
+$('article').fadeIn();
 `        $('article.template').hide();
-        }
-       $('#author-filter').val('');
-     });
-   };
-
-   DONE: Remove the setTeasers method,
-    and replace with a plain ole link in the article template.
-   articleView.setTeasers = function() {
-     $('.article-body *:nth-of-type(n+2)').hide();
-
-     $('#articles').on('click', 'a.read-on', function(e) {
-       e.preventDefault();
-       $(this).parent().find('*').fadeIn();
-       $(this).hide();
-     });
-   }; */
-
-  // COMMENT: What does this method do?  What is it's execution path?
+}
+$('#author-filter').val('');
+});
+};
+
+DONE: Remove the setTeasers method,
+and replace with a plain ole link in the article template.
+articleView.setTeasers = function() {
+$('.article-body *:nth-of-type(n+2)').hide();
+
+$('#articles').on('click', 'a.read-on', function(e) {
+e.preventDefault();
+$(this).parent().find('*').fadeIn();
+$(this).hide();
+});
+}; */
+
+// COMMENT: What does this method do?  What is it's execution path?
+// This method is mapping over the all articles array to render it all to the home page. At the same time, there are other methods being called that filters to hide any duplicates from being shown.
```

Comment:
This method is kind of the meat of our app! It is called in maaaaany of our routes after we've attached the appropriate articles to our ctx object.

This method first shows the articles section and hides all other primary sections (such as the about-me and admin sections). It empties out the articles section of all articles and takes the articles passed in (many times from the ctx.articles property, depending on where this method is being called) and appends them to the page after rendering them (which compiles html template, adds some properties, and returns the compiled template). It then calls articleView.populateFilters and articleView.handleFilters to append authors and categories to the appropriate select boxes and attach an event listener to those select boxes. It then truncates articles if they are longer than 1 paragraph.